sandbox/concurrent_bert_etl/app_all_lambdas/jobs.py

```diff
@@ -0,0 +1,303 @@
+
+    # Example event:
+    # {
+    #   "id": "tess-s0001-4-1",
+    #   "ra": 63.3739396231274,
+    #   "dec": -69.226822697583,
+    #   "radius": 20,
+    #   "bright_pixel_threshold": 1.9
+    # }
+    #
+    # work_queue populated by calling Lambda
+    for event in work_queue:
+        tess_id = event['id']
+        ra = float(event['ra'])
+        dec = float(event['dec'])
+        basename = f'{tess_id}_s3_uris.txt'
+        filename = os.path.join(homedir, basename)
+
+        try:
+            # Check if URI list already cached.
+            ologger.info(f'Attempting to download {basename} from S3')
+            outbucket.download_file(
+                basename, filename,
+                ExtraArgs={"RequestPayer": "requester"})
+        except Exception:
+            # Find full frame dataset for the observation ID.
+            ologger.info('Started quering Observations...')
+            obs_table = Observations.query_criteria(obs_id=tess_id)
+            products = Observations.get_product_list(obs_table)
+            filtered = Observations.filter_products(
+                products, productSubGroupDescription="FFIC",
+                mrp_only=False)
+
+            # Use AWS S3 bucket to pull data from.
+            Observations.enable_cloud_dataset(verbose=False)
+            ologger.info('Started obtaining cloud URIs...')
+            t_start = time.time()
+            s3_urls = Observations.get_cloud_uris(
+                filtered, include_bucket=False)
+            t_end = time.time()
+            ologger.info(f'Got {len(s3_urls)} URIs in {t_end - t_start} s')
+
+            # Upload URI list to cache.
+            with open(filename, 'w') as fout:
+                for url in s3_urls:
+                    fout.write(url + os.linesep)
+            try:
+                outbucket.upload_file(
+                    filename, basename,
+                    ExtraArgs={"RequestPayer": "requester"})
+            except Exception as exc:
+                ologger.error(str(exc))
+            else:
+                ologger.info(f'Uploaded {basename} to S3')
+        else:
+            # Use cache if it exists.
+            with open(filename, 'r') as fin:
+                s3_urls = [url.strip() for url in fin.readlines()]
+            ologger.info(f'Read {len(s3_urls)} URIs from {basename}')
+        finally:
+            # Clean up
+            if os.path.exists(filename):
+                os.remove(filename)
+
+        # Find pixel coordinates from sky from first frame header.
+        key = s3_urls[0]
+        basename = key.split('/')[-1]
+        filename = os.path.join(homedir, basename)
+        ologger.info(f'Resolving WCS from {key}')
+        bucket.download_file(
+            key, filename, ExtraArgs={"RequestPayer": "requester"})
+        hdr = fits.getheader(filename, ext=1)
+        w = WCS(hdr)
+        pix = w.all_world2pix(ra, dec, 0)
+        xpos = round(float(pix[0]))  # float needed to get rid of 0-D array
```

Comment:
TODO: "Let’s say the star needs to be at least 2*radii pixels away in both x and y." Otherwise, stop because too near the edge.